src/common/requests.ts

```diff
@@ -437,6 +438,29 @@ function makeRequest(
         /*initializer*/ ts.factory.createIdentifier("payload")
       )
     );
+    if ("content" in item.requestBody) {
```

Comment:
One thing that is not being checked here is if the requestBody is a `reference` object instead of the actual object, if it's a reference then the reference needs to be loaded, or else this will fail when spec's use references instead of inline objects. Please use the `isReferenceObject` util that's available, and load the reference if needed to cover all the bases

Reply:
using instance of or checking a discrimnator value ?

Reply:
```ts
if ("$ref" in item.requestBody) {
  // we have a reference that must be loaded
  var requestBody = $refs.get(item.requestBody.$ref)
  // continue with the same path of code as if it was inline
} else if ("content" in item.requestBody) {
  // actual value so we don't need to load the ref
}
```

Reply:
Please cover the reference code path with a unit test too to verify everything is working

Reply:
> Please cover the reference code path with a unit test too to verify everything is working

<img src="https://media3.giphy.com/media/QfzMP70zmNQiDf5sGP/giphy.gif"/>
i'm lost chief

Reply:
Add a unit test that tests the reference loading logic

Reply:
oh i see, i couldnt have figured that out myself 😓